apps/src/bin/quiche-server.rs

```diff
@@ -187,129 +217,114 @@ fn main() {
-            // is no connection matching, create a new one.
-            let client = if !clients_ids.contains_key(&hdr.dcid) &&
-                !clients_ids.contains_key(&conn_id)
-            {
-                if hdr.ty != quiche::Type::Initial {
-                    error!("Packet is not Initial");
-                    continue 'read;
+                    if let Ok(f) = std::fs::File::create(path) {
+                        let mut f = std::io::BufWriter::new(f);
+                        f.write_all(pkt_buf).ok();
+                    }
                 }
 
-                if !quiche::version_is_supported(hdr.version) {
-                    warn!("Doing version negotiation");
-
-                    let len =
-                        quiche::negotiate_version(&hdr.scid, &hdr.dcid, &mut out)
-                            .unwrap();
-
-                    let out = &out[..len];
+                pkt_count += 1;
 
-                    if let Err(e) = socket.send_to(out, from) {
-                        if e.kind() == std::io::ErrorKind::WouldBlock {
-                            trace!("send() would block");
-                            break;
-                        }
+                // Parse the QUIC packet's header.
+                let hdr = match quiche::Header::from_slice(
+                    pkt_buf,
+                    quiche::MAX_CONN_ID_LEN,
+                ) {
+                    Ok(v) => v,
 
-                        panic!("send() failed: {:?}", e);
-                    }
-                    continue 'read;
-                }
+                    Err(e) => {
+                        error!("Parsing packet header failed: {:?}", e);
+                        continue 'read;
+                    },
+                };
 
-                let mut scid = [0; quiche::MAX_CONN_ID_LEN];
-                scid.copy_from_slice(&conn_id);
+                trace!("got packet {:?}", hdr);
 
-                let mut odcid = None;
+                let conn_id = if !cfg!(feature = "fuzzing") {
+                    let conn_id = ring::hmac::sign(&conn_id_seed, &hdr.dcid);
+                    let conn_id = &conn_id.as_ref()[..quiche::MAX_CONN_ID_LEN];
+                    conn_id.to_vec().into()
+                } else {
+                    // When fuzzing use an all zero connection ID.
+                    [0; quiche::MAX_CONN_ID_LEN].to_vec().into()
+                };
 
-                if !args.no_retry {
-                    // Token is always present in Initial packets.
-                    let token = hdr.token.as_ref().unwrap();
+                // Lookup a connection based on the packet's connection ID. If
+                // there is no connection matching, create a new
+                // one.
+                let client = if !clients_ids.contains_key(&hdr.dcid) &&
+                    !clients_ids.contains_key(&conn_id)
+                {
+                    trace!("A new connection!");
```

Comment:
If we want to keep a log line for this, lets change what we print, for example to state the ID of the new connection, the source IP or something. It might be interesting to upgrade it to info level too

Reply:
Further down we have
`trace!("New connection: dcid={:?} scid={:?}", hdr.dcid, scid);`
I have changed that trace!() to:
`info!("A new connection has occurred. dcid={:?} scid={:?}", hdr.dcid, scid);`
and removed:
`trace!("A new connection!");`
I have also updated a later trace!() logging repeat connections to log out:
`info!("A repeat connection has occurred. Client ID={:?}", client_id);`